File: java/baiduads-sdk/src/main/java/com/baidu/dev2/api/sdk/manual/oauth/model/AccessTokenInfo.java
/*
 *  Baidu.com Inc.
 *  Copyright (c)  2021 All Rights Reserved.
 */

package com.baidu.dev2.api.sdk.manual.oauth.model;


import java.util.Map;

/**
 * @author fanliyan
 * @version 1.0
 * @since  2021/9/9 4:55 下午
 */
public class AccessTokenInfo {
    /**
     * 授权令牌
     */
    private String accessToken;

    /**
     * 更新令牌
     */
    private String refreshToken;

    /**
     * 授权令牌到期时间
     */
    private String expiresTime;

    /**
     * 更新令牌到期时间
     */
    private String refreshExpiresTime;

    /**
     * 授权令牌剩余有效时间，单位：秒
     */
    private Integer expiresIn;

    /**
     * 更新令牌剩余有效时间，单位：秒
     */
    private Integer refreshExpiresIn;

    /**
     * 应用的权限列表，客户关注value即可
     */
    private Map<String, String> scope;

    /**
     * 授权用户在应用+账户ID维度下对应的唯一标识
     */
    private String openId;

    /**
     * 授权账户ID
     */
    private Long userId;

    public String getAccessToken() {
        return accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public void setRefreshToken(String refreshToken) {
        this.refreshToken = refreshToken;
    }

    public String getExpiresTime() {
        return expiresTime;
    }

    public void setExpiresTime(String expiresTime) {
        this.expiresTime = expiresTime;
    }

    public String getRefreshExpiresTime() {
        return refreshExpiresTime;
    }

    public void setRefreshExpiresTime(String refreshExpiresTime) {
        this.refreshExpiresTime = refreshExpiresTime;
    }

    public Integer getExpiresIn() {
        return expiresIn;
    }

    public void setExpiresIn(Integer expiresIn) {
        this.expiresIn = expiresIn;
    }

    public Integer getRefreshExpiresIn() {
        return refreshExpiresIn;
    }

    public void setRefreshExpiresIn(Integer refreshExpiresIn) {
        this.refreshExpiresIn = refreshExpiresIn;
    }

    public Map<String, String> getScope() {
        return scope;
    }

    public void setScope(Map<String, String> scope) {
        this.scope = scope;
    }

    public String getOpenId() {
        return openId;
    }

    public void setOpenId(String openId) {
        this.openId = openId;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }
}

File: python/baiduads-sdk/baiduads_manual/oauth/model/access_token_info.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Copyright (c) 2022 Baidu.com, Inc. All Rights Reserved
"""


class AccessTokenInfo:
    """access token data model

    Attributes:
        accessToken: 授权令牌
        refreshToken: 刷新令牌
        expiresTime: 授权令牌到期时间
        refreshExpiresTime: 更新令牌到期时间
        expiresIn: 授权令牌剩余有效时间
        refreshExpiresIn: 更新令牌剩余有效时间
        scope: 使用应用的用户已授权的权限列表
        userId: 授权账号 ucid
    """

    def __init__(self,
                 accessToken, refreshToken, expiresTime,
                 refreshExpiresTime, expiresIn, refreshExpiresIn,
                 scope, userId):
        self.accessToken = accessToken
        self.refreshToken = refreshToken
        self.expiresTime = expiresTime
        self.refreshExpiresTime = refreshExpiresTime
        self.expiresIn = expiresIn
        self.refreshExpiresIn = refreshExpiresIn
        self.scope = scope
        self.userId = userId

    @classmethod
    def dict_to_obj(cls, dict_data):
        """ 用于将dict转换为对象
        """
        return AccessTokenInfo(
            dict_data.get("accessToken", None),
            dict_data.get("refreshToken", None),
            dict_data.get("expiresTime", None),
            dict_data.get("refreshExpiresTime", None),
            dict_data.get("expiresIn", None),
            dict_data.get("refreshExpiresIn", None),
            dict_data.get("scope", None),
            dict_data.get("userId", None)
        )
